src/doctopus/storage.clj

```diff
@@ -0,0 +1,46 @@
+(ns doctopus.storage
+  "Defines how Doctopus speaks to different places it can store generated HTML"
+  (:require [doctopus.storage-impls :as storage-impls]))
+
+;; ## Storage
+;; Doctopus shouldn't really care _where_ it puts generated HTML; it
+;; just needs a consistent interface. Thus, the Backend: a way of
+;; generalizing and controlling where Doctopus tries to get and put Stuff.
+;;
+;; As a first pass, the backend will be defined by an atom which
+;; contains the current Backend instance; functions in this NS will
+;; work against that atom. This means a Doctopus will have a single Backend; we
+;; can potentially expand this later.
+;;
+;; Initial backends:
+;; - Temporary filesystem (provides it's own directory)
+;; - Permanent filesystem (needs a directory provided)
+;; - Amazon S3 (maybe?)
+;;
+;; At time of writing, I'm thinking of loading and saving one Tentacle worth of
+;; documents all at once.
+
+(defprotocol DoctopusBackend
+  (set-backend! [this backend])
+  (load-from-storage [this k])
+  (save-to-storage [this k v]))
```

Comment:
Thiiiis is an interesting point. Probably we want that for the permanent storage system, yes. Hrm.